keras/layers/recurrent.py

```diff
@@ -198,6 +214,16 @@ def get_initial_states(self, x):
         initial_states = [initial_state for _ in range(len(self.states))]
         return initial_states
 
+    def bn(self, x, _gamma, _beta, matrix_amount=1):
+        _gamma = K.expand_dims(_gamma, dim=0)
+        _beta = K.expand_dims(_beta, dim=0)
+        x = K.reshape(x, [-1, matrix_amount, self.output_dim])
+        x_var = x - K.mean(x, axis=2, keepdims=True)
+        x_std = K.sqrt(K.mean(K.square(x_var), axis=2, keepdims=True) + self.epsilon)
+        res = x_var / x_std * _gamma + _beta
```

Comment:
This is per-batch normalization, usually used during training. You are not computing running averages of std and mean for normalization during testing. Why?